### DISCUSSION TITLE
Huge Huge HTML size when using repeater

### DISCUSSION THREAD

**Discussion by ariaieboy** [filamentphp/filament#8706] [Help]
Hello everyone,
I know that the document said that we should not use Repeater for big forms and relationships but In my case the Repeater is the best option.
I have a Repeater inside anotherRepeater like this:
```php
Forms\Components\Repeater::make('groups')
                    ->itemLabel(fn (array $state): ?string => $state['title'] ?? null)
                    ->relationship()
                    ->defaultItems(0)
                    ->reorderable()
                    ->schema([
                        Forms\Components\TextInput::make('title')->required(),
                        Forms\Components\Repeater::make('items')
                            ->relationship()
                            ->collapsed()
                            ->reorderable()
                            ->reorderableWithButtons()
                            ->defaultItems(0)
                            ->itemLabel(fn (array $state): string => ($state['title'] ?? null).' - '.(number_format($state['price'])))
                            ->schema([
                                Forms\Components\Tabs::make('items')
                                    ->tabs([
                                        Forms\Components\Tabs\Tab::make('tab1')
                                            ->schema([
                                                Forms\Components\TextInput::make('title')->required(),
                                                Forms\Components\TextInput::make('price')->default(0)->required()->currencyMask(),
                                            ])->columns(2),
                                        Forms\Components\Tabs\Tab::make('tab2')->schema([
                                            Forms\Components\Textarea::make('description'),
                                            Forms\Components\Select::make('product_id')->preload()->searchable()->relationship('product', 'title_fa'),
                                            Forms\Components\Toggle::make('status')->default(true),
                                        ]),
                                    ]),
                            ])->reorderableWithButtons(),
                    ])->columnSpanFull()->reorderableWithButtons()->reorderableWithDragAndDrop(false)->collapsed(),
```

The problem is that my HTML file size get too big that Nginx returns Err::http2_protocol_err

for example a record containing 3 groups and 9 items generates a page with 5.2M response size 😱
![image](https://github.com/filamentphp/filament/assets/15873972/f982cb09-dfbd-422b-bbe3-2e104b5440af)

**Comment by ariaieboy** [comment]
@danharrin is this a bug or it's normal? 
should I create an issue for this?

**Reply by krekas** [reply]
It should be how livewire works and renders full page components. Also, I think you are using chrome browser which doesn't show compressed size that is sent to browser.

**Reply by ariaieboy** [reply]
@krekas I don't think so I have lots of enormous livewire components in my application but none of them exceed 100KB HTML size.
One problem here is that filament uses inline SVG instead of loading it from a URL. Since we have `reorderableWithButtons` for each item, it adds lots of inline SVGs.

**Reply by krekas** [reply]
Define what is enormous livewire component. If it's logic wise that doesn't count. It is whats renderable.

**Comment by ariaieboy** [comment]
![image](https://github.com/filamentphp/filament/assets/15873972/81b39357-4997-49c6-882d-61a3962b6f22)

According to the https://www.debugbear.com/html-size-analyzer 86.82% of the page size is related to the divs with x-data 🤔
